mteb/abstasks/AbsTaskMultilabelClassification.py

```diff
@@ -0,0 +1,146 @@
+        self.batch_size = batch_size
+
+        # Bootstrap parameters
+        self.n_experiments = (
+            n_experiments
+            if n_experiments is not None
+            else self.metadata_dict.get("n_experiments", 10)
+        )
+        self.samples_per_label = (
+            samples_per_label
+            if samples_per_label is not None
+            else self.metadata_dict.get("samples_per_label", 8)
+        )
+        # kNN parameters
+        self.k = k
+        # Run metadata validation by instantiating addressing the attribute
+        # This is quite hacky. Ideally, this would be done in the constructor of
+        # each concrete task, but then we have to duplicate the __init__ method's
+        # interface.
+        if hasattr(self, "metadata"):
+            self.metadata
+
+    def _add_main_score(self, scores):
+        if self.metadata_dict["main_score"] in scores:
+            scores["main_score"] = scores[self.metadata_dict["main_score"]]
+        else:
+            logger.warn(
+                f"main score {self.metadata_dict['main_score']} not found in scores {scores.keys()}"
+            )
+
+    def evaluate(self, model, eval_split="test", train_split="train", **kwargs):
+        if not self.data_loaded:
+            self.load_data()
+
+        if self.is_multilingual:
+            scores = {}
+            for lang in self.dataset:
+                logger.info(
+                    f"\nTask: {self.metadata_dict['name']}, split: {eval_split}, language: {lang}. Running..."
+                )
+                scores[lang] = self._evaluate_monolingual(
+                    model, self.dataset[lang], eval_split, train_split, **kwargs
+                )
+                self._add_main_score(scores[lang])
+        else:
+            logger.info(
+                f"\nTask: {self.metadata_dict['name']}, split: {eval_split}. Running..."
+            )
+            scores = self._evaluate_monolingual(
+                model, self.dataset, eval_split, train_split, **kwargs
+            )
+            self._add_main_score(scores)
+
+        return scores
+
+    def _evaluate_monolingual(
+        self, model, dataset, eval_split="test", train_split="train", **kwargs
+    ):
+        train_split = dataset[train_split]
+        eval_split = dataset[eval_split]
+        params = {"k": self.k, "batch_size": self.batch_size}
+        params.update(kwargs)
+
+        scores = []
+        test_cache, idxs = (
+            None,
+            None,
+        )  # we store idxs to make the shuffling reproducible
+        for i in range(self.n_experiments):
+            logger.info(
+                "=" * 10 + f" Experiment {i+1}/{self.n_experiments} " + "=" * 10
+            )
+            # Bootstrap `self.samples_per_label` samples per label for each split
+            X_sampled, y_sampled, idxs = self._undersample_data(
+                train_split["text"], train_split["label"], self.samples_per_label, idxs
+            )
```

Comment:
You could essentially sample the same and embed it twice which seems slightly inefficient, which is slightly annoying. Another solution would be so sample the ids first, embed them all at once, and then sample the embeddings. This also leads to less task switching.